src/autoscaler/scalingengine/schedule/schedule_test.go

```diff
@@ -0,0 +1,240 @@
+		activeSchedule = ActiveSchedule{
+			ScheduleId:      "a-schedule-id",
+			InstanceInitial: 5,
+			InstanceMin:     3,
+			InstanceMax:     10,
+		}
+	})
+
+	Describe("SetActiveSchedule", func() {
+		JustBeforeEach(func() {
+			err = schedules.SetActiveSchedule("an-app-id", activeSchedule)
+		})
+
+		Context("when no active schedule exists for the app", func() {
+			It("sets the active schedule and initial instances", func() {
+				Expect(err).NotTo(HaveOccurred())
+				appId, instances := cfc.SetAppInstancesArgsForCall(0)
+				Expect(appId).To(Equal("an-app-id"))
+				Expect(instances).To(Equal(5))
+
+				schedule, exists := schedules.GetActiveSchedule("an-app-id")
+				Expect(exists).To(BeTrue())
+				Expect(schedule).To(Equal(activeSchedule))
+			})
+
+		})
+
+		Context("when active schedule exists for the app", func() {
+			BeforeEach(func() {
+				err = schedules.SetActiveSchedule("an-app-id", ActiveSchedule{
+					ScheduleId:      "currrent-schedule-id",
+					InstanceInitial: 6,
+					InstanceMin:     2,
+					InstanceMax:     8,
+				})
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			It("Sets the new active scheule and logs the info", func() {
+				Expect(err).NotTo(HaveOccurred())
+				schedule, exists := schedules.GetActiveSchedule("an-app-id")
+				Expect(exists).To(BeTrue())
+				Expect(schedule).To(Equal(activeSchedule))
+
+				appId, instances := cfc.SetAppInstancesArgsForCall(1)
+				Expect(appId).To(Equal("an-app-id"))
+				Expect(instances).To(Equal(5))
+
+				Eventually(buffer).Should(gbytes.Say("active schedule exists"))
+			})
+		})
+
+		Context("when fails to set initial instances number", func() {
+			BeforeEach(func() {
+				cfc.SetAppInstancesReturns(errors.New("test error"))
+			})
+			It("should sets the active schedule and return an error", func() {
+				Expect(err).To(HaveOccurred())
+
+				schedule, exists := schedules.GetActiveSchedule("an-app-id")
+				Expect(exists).To(BeTrue())
+				Expect(schedule).To(Equal(activeSchedule))
+
+				Eventually(buffer).Should(gbytes.Say("failed-set-active-schedule-set-instance-initial"))
+				Eventually(buffer).Should(gbytes.Say("test error"))
+			})
+		})
+	})
+
+	Describe("RemoveActiveSchedule", func() {
+		JustBeforeEach(func() {
+			err = schedules.RemoveActiveSchedule("an-app-id", "a-schedule-id")
+		})
+
+		Context("when the active schedule exists", func() {
+			BeforeEach(func() {
+				err = schedules.SetActiveSchedule("an-app-id", activeSchedule)
+				Expect(err).NotTo(HaveOccurred())
+				policyDB.GetAppPolicyReturns(&models.ScalingPolicy{}, nil)
+			})
+
+			It("removes the active schedule", func() {
+				Expect(err).NotTo(HaveOccurred())
+				_, exists := schedules.GetActiveSchedule("an-app-id")
+				Expect(exists).To(BeFalse())
+			})
+		})
+
+		Context("when  active schedule does not exist", func() {
+			BeforeEach(func() {
+				policyDB.GetAppPolicyReturns(&models.ScalingPolicy{}, nil)
+			})
+
+			It("logs the information", func() {
```

Comment:
testing logging doesn't test the function.
It("does not error") is what you care about.